docs/reference/scripting/using.asciidoc

```diff
@@ -197,6 +197,86 @@ them in other cases to remove ambiguity.
 Use this abbreviated syntax anywhere that {es} supports scripts, such as
 when you're creating <<runtime-mapping-fields,runtime fields>>.
 
+[discrete]
+[[script-fields-api]]
+=== Access document fields
+The examples from previous sections use the `field` API to access document
+fields:
+
+[source,painless]
+----
+field('my_field').get(null)
+----
+
+The `field` API fundamentally changes how you access documents in Painless.
+Previously, you had to declare `doc` and the field name that you wanted to
+access:
+
+[source,painless]
+----
+doc['my_field'].value
+----
+
+Accessing document fields this way didn't handle missing values, which meant
+that your Painless scripts had to check that fields exist and create
+intermediary objects for handling such omissions.
+
+Instead, use the `field` API, which is the preferred approach to access
+documents in Painless. The `field` API abstracts access options (such as
+`_source` or `doc_values`) and provides a field-centric view of your documents.
+
+The `field` API always returns a non-null object that represents the given field,
```

Comment:
I think we need to differentiate fields that return primitive types versus fields that return Object types. `non-null` isn't correct if the default is `null`.